src/prefect/new_task_engine.py

```diff
@@ -167,8 +168,21 @@ def _hook_fn():
     def _compute_state_details(
         self, include_cache_expiration: bool = False
     ) -> StateDetails:
+        from prefect.utilities.engine import should_log_prints
+
+        # We sometimes call this prior to setting up the run context, so we create a temporary
+        # context here
+        # TODO: Should we set up the run context earlier in Engine.start?
+        log_prints = should_log_prints(self.task)
+        task_run_context = TaskRunContext(
+            task=self.task,
+            log_prints=log_prints,
+            task_run=self.task_run,
+            parameters=self.parameters,
```

Comment:
This is a hack to get the tests to pass. Should we move the setup of the task run context before calling `Engine.begin_run`?

Reply:
Yep this definitely feels out of place to me too.  I think you're right about moving it earlier so we don't have to do this here